mindsdb_native/libs/data_types/data_source.py

```diff
@@ -66,7 +108,71 @@ def drop_columns(self, column_list):
             else:
                 columns_to_drop.append(self._col_map[col])
 
-        self.df.drop(columns=columns_to_drop, inplace=True)
+        self._internal_df.drop(columns=columns_to_drop, inplace=True)
+
+    def _filter_to_pandas(self, raw_condition, df):
+        """Convert filter conditions to a paticular
+        DataFrame instance"""
+        mapping = {
+                    ">": lambda x, y: df[x] > y,
+                    "like": lambda x, y: df[x].str.contains(y.replace("%", "")),
+                    "<": lambda x, y: df[x] < y,
+                    "=": lambda x, y: df[x] == y,
+                    "!=": lambda x, y: df[x] != y
+                  }
+        col, cond, val = raw_condition
+        return mapping[cond.lower()](col, val)
+
+    def filter(self, where=None, limit=None, get_col_map=False):
+        """Convert SQL like filter requests to pandas DataFrame filtering"""
+        if self.is_sql:
+            parsed_query = moz_sql_parser.parse(self.query)
+
+            for col, op, value in where or []:
+                past_where_clause = parsed_query.get('where', {})
+
+                op = op.lower()
+                op_json = binary_ops.get(op, None)
+                if op_json is None:
+                    log.warning(f"Operator: {op} not found in: QueryBuilder._OPERATORS\n Using it anyway.")
+                    op_json = op.lower()
+
+                if op.lower() == 'like':
+                    value = '%' + value.strip('%') + '%'
+
+                where_clause = {op_json: [col, value]}
+
+                if len(past_where_clause) > 0:
+                    where_clause = {'and': [where_clause, past_where_clause]}
+
+                parsed_query['where'] = where_clause
+
+            if limit is not None:
+                parsed_query['limit'] = limit
+
+            query = moz_sql_parser.format(parsed_query)
+            query = query.replace('"',"'")
+
+            args = deepcopy(self.args)
+            kwargs = deepcopy(self.kwargs)
+
+            if 'query' in kwargs:
+                kwargs['query'] = query
+            else:
+                args[0] = query
+
+            if get_col_map:
+                return self._setup(*args, **kwargs)
+            else:
+                return self._setup(*args, **kwargs)[0]
```

Comment:
I'm afraid this will only work in simple cases, because:

1. you can parse only simple select:
```
p.parse("SELECT row_number() OVER (ORDER BY id) as nom FROM TABLE")
error
```

2. back query assemble possible not in any case:
```
p.format(p.parse(" with a as (select b from c) select * from a"))
'SELECT * FROM a'
```

3. Parser dont know about syntax features of databases. In mysql insead 'LIMIT x OFFSET y' user can use OFFSET x, y:
```
p.parse("select a from b limit 5, 5")
error
```

4. You will loose escaping on back assemble:
```
p.format(p.parse("select a from `b b` "))
'SELECT a FROM "b b"'
```

after it, you use `query = query.replace('"',"'")`, so we get transform:
```
select a from `b b`  ->  'SELECT a FROM 'b b'
```

Reply:
Most queries will be simple cases, but yeah, we should have a fallback where if the query fails we filter on the data frame.

As for the escaping, that is an issue I have to deal with better :(

Reply:
Ok, I added fallback to  data frame filtering when the query fails to parse or to execute, and I added 2x new test scenarios to validate this.

In the process of doing this I also made some adjustments to data frame filtering.